stablehlo/dialect/Base.cpp

```diff
@@ -600,5 +600,23 @@ ShapedType createShapedType(ShapedTypeComponents components) {
   return UnrankedTensorType::get(components.getElementType());
 }
 
+// TODO(#1578): Remove this code once all uses of I64DenseArrayOrElements1DAttr
+// have been removed.
+SmallVector<int64_t> i64ArrayOrElementsValues(Attribute attr) {
+  SmallVector<int64_t> vec;
+  if (!attr) return vec;
+  if (auto elements = dyn_cast_or_null<DenseIntElementsAttr>(attr)) {
+    vec = llvm::to_vector(elements.getValues<int64_t>());
```

Comment:
- No ned for `else if` statements, each if can just return.
- No need for temporary vector
- I think we can usre `dyn_cast` here instead of `dyn_cast_or_null`
- Attributes have a `dyn_cast` method
- Why the `false` arg to `report_fatal_error`?

All together that makes for:

```cpp
if (!attr) return {};
if (auto elements = attr.dyn_cast<DenseIntElementsAttr>())
  return llvm::to_vector(elements.getValues<int64_t>());
if (auto array = attr.dyn_cast<DenseI64ArrayAttr>())
  return llvm::to_vector(array.asArrayRef());
llvm::report_fatal_error(
    "called i64ArrayOrElementsValues on Attribute that was neither a "
    "DenseIntElementsAttr or a DenseI64ArrayAttr");
```